### PR TITLE
Fix orgnization user watch repository

### LINKED ISSUE #2667: Mirrors Repo BUG
## Configures

- Gitea version: 1.2
- Git version: 2.14.2
- Operating system: CentOS 7.3 x64
- Database:
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [X] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [X] Yes ( https://try.gitea.io/supertest/installer-framework/watchers )
  - [ ] No
  - [ ] Not relevant

## Descriptions

1. Create a organization ( test: https://try.gitea.io/supertest/ );
2. New Migration for organization ( test: https://try.gitea.io/supertest/installer-framework );
3. See watchers ( test: https://try.gitea.io/supertest/installer-framework/watchers ).

## Others

**This bug also appeared in Gogs and latest commit version of Gitea.**

### PR DESCRIPTION
**Pull request by lunny**
fix #2667

### CODE REVIEW

**Comment by lafriks** [comment]
Does user creating repository for organization need to be added automatically as watcher?

File: models/repo.go [hunk 1]
```diff
@@ -1295,7 +1295,7 @@ func createRepository(e *xorm.Session, doer, u *User, repo *Repository) (err err
 		}
 	}
 
-	if err = watchRepo(e, u.ID, repo.ID, true); err != nil {
+	if err = watchRepo(e, doer.ID, repo.ID, true); err != nil {
```
Comment (ethantkoenig) [inline models/repo.go:1298]: Need to update other call to `watchRepo(..)` on line 1483

File: models/migrations/migrations.go [hunk 2]
```diff
@@ -140,6 +140,8 @@ var migrations = []Migration{
 	NewMigration("remove duplicate unit types", removeDuplicateUnitTypes),
 	// v45 -> v46
 	NewMigration("remove index column from repo_unit table", removeIndexColumnFromRepoUnitTable),
+	// v46 -> v47
+	NewMigration("remove orgnization watch repositories", removeOrgnizationWatchRepo),
```
Comment (ethantkoenig) [inline models/migrations/migrations.go]: spelling (orgnization -> organization), both the comment and function name.

**Review by ethantkoenig — CHANGES_REQUESTED** [review]

**Comment by lunny** [comment]
@lafriks I think yes.

File: models/repo.go (hunk 1 above)
Comment (lunny) [reply models/repo.go:1298]: done.

File: models/migrations/migrations.go (hunk 2 above)
Comment (lunny) [reply models/migrations/migrations.go]: done.

**Review by ethantkoenig — APPROVED** [review]

**Comment by ethantkoenig** [comment]
LGTM

Would it be worth changing `watchRepo(..)` to take a user argument, instead of a user-id, and check that the user is an individual? Right now, I think there is no way for the user-id to be an organization, but it might protect against future bugs. Just a thought.

**Comment by lunny** [comment]
@ethantkoenig Good idea. But for keeping the PR simple, I think that could be another PR.

File: models/migrations/v46.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+package migrations
+
+import (
+	"github.com/go-xorm/xorm"
+)
+
+func removeOrganizationWatchRepo(x *xorm.Engine) error {
+	// Watch is connection request for receiving repository notification.
+	type Watch struct {
```
Comment (Morlinest) [inline models/migrations/v46.go]: Not used, can be removed.

**Review by Morlinest — CHANGES_REQUESTED** [review]

**Comment by lunny** [comment]
@Morlinest fixed and rebased.

**Review by Morlinest — APPROVED** [review]

**Comment by Morlinest** [comment]
LGTM

**Pull request merged**